### ISSUE TITLE
Bug: Missing dependency - requests

### ISSUE THREAD

**Issue by deenine** [oscal-compass/compliance-trestle#1487]
## Describe the bug

Installing trestle into a new venv per [install instructions](https://oscal-compass.github.io/compliance-trestle/python_trestle_setup/) results in inoperable trestle, due to missing requests dependency. 

## To Reproduce

```
trestle_tings % rm -rf venv.trestle 
trestle_tings % python3 -m venv venv.trestle            
trestle_tings % . venv.trestle/bin/activate             
(venv.trestle)  trestle_tings % python3 --version                       
Python 3.9.6
(venv.trestle)  trestle_tings % pip --version                           
pip 21.2.4 from /Users/user/Workspace/trestle_tings/venv.trestle/lib/python3.9/site-packages/pip (python 3.9)
(venv.trestle)  trestle_tings % python -m pip install compliance-trestle
Collecting compliance-trestle
  Using cached compliance_trestle-2.4.0-py2.py3-none-any.whl (422 kB)
....
Installing collected packages: multidict, idna, frozenlist, yarl, typing-extensions, dnspython, attrs, async-timeout, aiosignal, tomli, sniffio, pydantic, pycparser, platformdirs, pathspec, packaging, mypy-extensions, MarkupSafe, h11, exceptiongroup, email-validator, click, certifi, aiohttp, toml, six, pyyaml, Jinja2, isort, inflect, httpcore, genson, cffi, black, argcomplete, anyio, ruamel.yaml.clib, pynacl, orderedmultidict, httpx, et-xmlfile, datamodel-code-generator, cryptography, bcrypt, ruamel.yaml, python-frontmatter, python-dotenv, paramiko, orjson, openpyxl, ilcli, furl, defusedxml, cmarkgfm, compliance-trestle
Successfully installed Jinja2-3.1.2 MarkupSafe-2.1.3 aiohttp-3.9.1 aiosignal-1.3.1 anyio-4.2.0 argcomplete-3.2.1 async-timeout-4.0.3 attrs-23.1.0 bcrypt-4.1.2 black-24.1a1 certifi-2023.11.17 cffi-1.16.0 click-8.1.7 cmarkgfm-0.6.0 compliance-trestle-2.4.0 cryptography-41.0.7 datamodel-code-generator-0.25.2 defusedxml-0.7.1 dnspython-2.4.2 email-validator-2.1.0.post1 et-xmlfile-1.1.0 exceptiongroup-1.2.0 frozenlist-1.4.1 furl-2.1.3 genson-1.2.2 h11-0.14.0 httpcore-1.0.2 httpx-0.26.0 idna-3.6 ilcli-0.3.2 inflect-5.6.2 isort-5.13.2 multidict-6.0.4 mypy-extensions-1.0.0 openpyxl-3.1.2 orderedmultidict-1.0.1 orjson-3.9.10 packaging-23.2 paramiko-3.4.0 pathspec-0.12.1 platformdirs-4.1.0 pycparser-2.21 pydantic-1.10.13 pynacl-1.5.0 python-dotenv-1.0.0 python-frontmatter-1.0.1 pyyaml-6.0.1 ruamel.yaml-0.18.5 ruamel.yaml.clib-0.2.8 six-1.16.0 sniffio-1.3.0 toml-0.10.2 tomli-2.0.1 typing-extensions-4.9.0 yarl-1.9.4
WARNING: You are using pip version 21.2.4; however, version 23.3.2 is available.
You should consider upgrading via the '/Users/user/Workspace/trestle_tings/venv.trestle/bin/python -m pip install --upgrade pip' command.
(venv.trestle)  trestle_tings % trestle --help
Traceback (most recent call last):
  File "/Users/user/Workspace/trestle_tings/venv.trestle/bin/trestle", line 5, in <module>
    from trestle.cli import run
  File "/Users/user/Workspace/trestle_tings/venv.trestle/lib/python3.9/site-packages/trestle/cli.py", line 24, in <module>
    from trestle.core.commands.assemble import AssembleCmd
  File "/Users/user/Workspace/trestle_tings/venv.trestle/lib/python3.9/site-packages/trestle/core/commands/assemble.py", line 23, in <module>
    from trestle.common.load_validate import load_validate_model_path
  File "/Users/user/Workspace/trestle_tings/venv.trestle/lib/python3.9/site-packages/trestle/common/load_validate.py", line 25, in <module>
    from trestle.common.model_utils import ModelUtils
  File "/Users/user/Workspace/trestle_tings/venv.trestle/lib/python3.9/site-packages/trestle/common/model_utils.py", line 38, in <module>
    from trestle.core.remote import cache
  File "/Users/user/Workspace/trestle_tings/venv.trestle/lib/python3.9/site-packages/trestle/core/remote/cache.py", line 37, in <module>
    import requests
ModuleNotFoundError: No module named 'requests'
```

## Expected behavior

Trestle help dialogue.

## Environment

- OS: macOS 14.1.2
- Python version: Python 3.9.6
- Installed packages:

**Comment by degenaro** [comment]
Works for me. Still fails for you??

degenaro:~$ cat /etc/redhat-release
Red Hat Enterprise Linux release 8.8 (Ootpa)
degenaro:~$ python --version
Python 3.9.9
$ python -m venv venv.trestle.pip
degenaro:~$ source venv.trestle.pip/bin/activate
(venv.trestle.pip) degenaro:~$ python -m pip install compliance-trestle
Collecting compliance-trestle
  Downloading compliance_trestle-2.4.0-py2.py3-none-any.whl (422 kB)
     |████████████████████████████████| 422 kB 2.7 MB/s 
Collecting cmarkgfm==0.6.*
  Using cached cmarkgfm-0.6.0-cp39-cp39-manylinux2010_x86_64.whl (446 kB)
Collecting pydantic[email]<2.0.0,>=1.8.2
  Using cached pydantic-1.10.13-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (3.2 MB)
Collecting furl
  Using cached furl-2.1.3-py2.py3-none-any.whl (20 kB)
Collecting attrs
  Downloading attrs-23.2.0-py3-none-any.whl (60 kB)
     |████████████████████████████████| 60 kB 14.8 MB/s 
Collecting orjson
  Downloading orjson-3.9.10-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (138 kB)
     |████████████████████████████████| 138 kB 49.0 MB/s 
Collecting paramiko
  Downloading paramiko-3.4.0-py3-none-any.whl (225 kB)
     |████████████████████████████████| 225 kB 73.2 MB/s 
Collecting datamodel-code-generator[http]>=0.11.14
  Downloading datamodel_code_generator-0.25.2-py3-none-any.whl (107 kB)
     |████████████████████████████████| 107 kB 66.2 MB/s 
Collecting python-dotenv>=0.10.4
  Using cached python_dotenv-1.0.0-py3-none-any.whl (19 kB)
Collecting python-frontmatter
  Downloading python_frontmatter-1.0.1-py3-none-any.whl (9.2 kB)
Collecting defusedxml
  Using cached defusedxml-0.7.1-py2.py3-none-any.whl (25 kB)
Collecting ruamel.yaml
  Downloading ruamel.yaml-0.18.5-py3-none-any.whl (116 kB)
     |████████████████████████████████| 116 kB 68.4 MB/s 
Collecting ilcli
  Using cached ilcli-0.3.2-py2.py3-none-any.whl (11 kB)
Collecting cryptography
  Downloading cryptography-41.0.7-cp37-abi3-manylinux_2_28_x86_64.whl (4.4 MB)
     |████████████████████████████████| 4.4 MB 74.8 MB/s 
Collecting openpyxl~=3.0
  Using cached openpyxl-3.1.2-py2.py3-none-any.whl (249 kB)
Collecting Jinja2>=3.0.1
  Using cached Jinja2-3.1.2-py3-none-any.whl (133 kB)
Collecting cffi>=1.0.0
  Using cached cffi-1.16.0-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (443 kB)
Collecting pycparser
  Using cached pycparser-2.21-py2.py3-none-any.whl (118 kB)
Collecting genson<2.0,>=1.2.1
  Using cached genson-1.2.2-py2.py3-none-any.whl
Collecting packaging
  Using cached packaging-23.2-py3-none-any.whl (53 kB)
Collecting inflect<6.0,>=4.1.0
  Using cached inflect-5.6.2-py3-none-any.whl (33 kB)
Collecting pyyaml>=6.0.1
  Using cached PyYAML-6.0.1-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (738 kB)
Collecting toml<1.0.0,>=0.10.0
  Using cached toml-0.10.2-py2.py3-none-any.whl (16 kB)
Collecting isort<6.0,>=4.3.21
  Downloading isort-5.13.2-py3-none-any.whl (92 kB)
     |████████████████████████████████| 92 kB 355 kB/s 
Collecting black>=19.10b0
  Downloading black-24.1a1-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (1.7 MB)
     |████████████████████████████████| 1.7 MB 37.9 MB/s 
Collecting argcomplete<4.0,>=1.10
  Downloading argcomplete-3.2.1-py3-none-any.whl (42 kB)
     |████████████████████████████████| 42 kB 2.1 MB/s 
Collecting httpx
  Downloading httpx-0.26.0-py3-none-any.whl (75 kB)
     |████████████████████████████████| 75 kB 7.2 MB/s 
Collecting platformdirs>=2
  Downloading platformdirs-4.1.0-py3-none-any.whl (17 kB)
Collecting tomli>=1.1.0
  Using cached tomli-2.0.1-py3-none-any.whl (12 kB)
Collecting aiohttp>=3.7.4
  Downloading aiohttp-3.9.1-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (1.2 MB)
     |████████████████████████████████| 1.2 MB 43.8 MB/s 
Collecting typing-extensions>=4.0.1
  Downloading typing_extensions-4.9.0-py3-none-any.whl (32 kB)
Collecting mypy-extensions>=0.4.3
  Using cached mypy_extensions-1.0.0-py3-none-any.whl (4.7 kB)
Collecting pathspec>=0.9.0
  Downloading pathspec-0.12.1-py3-none-any.whl (31 kB)
Collecting click>=8.0.0
  Using cached click-8.1.7-py3-none-any.whl (97 kB)
Collecting multidict<7.0,>=4.5
  Downloading multidict-6.0.4-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (114 kB)
     |████████████████████████████████| 114 kB 36.8 MB/s 
Collecting async-timeout<5.0,>=4.0
  Downloading async_timeout-4.0.3-py3-none-any.whl (5.7 kB)
Collecting aiosignal>=1.1.2
  Downloading aiosignal-1.3.1-py3-none-any.whl (7.6 kB)
Collecting yarl<2.0,>=1.0
  Downloading yarl-1.9.4-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (304 kB)
     |████████████████████████████████| 304 kB 52.8 MB/s 
Collecting frozenlist>=1.1.1
  Downloading frozenlist-1.4.1-cp39-cp39-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl (240 kB)
     |████████████████████████████████| 240 kB 46.6 MB/s 
Collecting MarkupSafe>=2.0
  Using cached MarkupSafe-2.1.3-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (25 kB)
Collecting et-xmlfile
  Using cached et_xmlfile-1.1.0-py3-none-any.whl (4.7 kB)
Collecting email-validator>=1.0.3
  Downloading email_validator-2.1.0.post1-py3-none-any.whl (32 kB)
Collecting dnspython>=2.0.0
  Using cached dnspython-2.4.2-py3-none-any.whl (300 kB)
Collecting idna>=2.0.0
  Downloading idna-3.6-py3-none-any.whl (61 kB)
     |████████████████████████████████| 61 kB 298 kB/s 
Collecting orderedmultidict>=1.0.1
  Using cached orderedmultidict-1.0.1-py2.py3-none-any.whl (11 kB)
Collecting six>=1.8.0
  Using cached six-1.16.0-py2.py3-none-any.whl (11 kB)
Collecting anyio
  Downloading anyio-4.2.0-py3-none-any.whl (85 kB)
     |████████████████████████████████| 85 kB 7.5 MB/s 
Collecting sniffio
  Using cached sniffio-1.3.0-py3-none-any.whl (10 kB)
Collecting httpcore==1.*
  Downloading httpcore-1.0.2-py3-none-any.whl (76 kB)
     |████████████████████████████████| 76 kB 12.1 MB/s 
Collecting certifi
  Downloading certifi-2023.11.17-py3-none-any.whl (162 kB)
     |████████████████████████████████| 162 kB 42.8 MB/s 
Collecting h11<0.15,>=0.13
  Using cached h11-0.14.0-py3-none-any.whl (58 kB)
Collecting exceptiongroup>=1.0.2
  Downloading exceptiongroup-1.2.0-py3-none-any.whl (16 kB)
Collecting bcrypt>=3.2
  Downloading bcrypt-4.1.2-cp39-abi3-manylinux_2_28_x86_64.whl (698 kB)
     |████████████████████████████████| 698 kB 94.3 MB/s 
Collecting pynacl>=1.5
  Using cached PyNaCl-1.5.0-cp36-abi3-manylinux_2_17_x86_64.manylinux2014_x86_64.manylinux_2_24_x86_64.whl (856 kB)
Collecting ruamel.yaml.clib>=0.2.7
  Using cached ruamel.yaml.clib-0.2.8-cp39-cp39-manylinux_2_5_x86_64.manylinux1_x86_64.whl (562 kB)
Installing collected packages: multidict, idna, frozenlist, yarl, typing-extensions, dnspython, attrs, async-timeout, aiosignal, tomli, sniffio, pydantic, pycparser, platformdirs, pathspec, packaging, mypy-extensions, MarkupSafe, h11, exceptiongroup, email-validator, click, certifi, aiohttp, toml, six, pyyaml, Jinja2, isort, inflect, httpcore, genson, cffi, black, argcomplete, anyio, ruamel.yaml.clib, pynacl, orderedmultidict, httpx, et-xmlfile, datamodel-code-generator, cryptography, bcrypt, ruamel.yaml, python-frontmatter, python-dotenv, paramiko, orjson, openpyxl, ilcli, furl, defusedxml, cmarkgfm, compliance-trestle
Successfully installed Jinja2-3.1.2 MarkupSafe-2.1.3 aiohttp-3.9.1 aiosignal-1.3.1 anyio-4.2.0 argcomplete-3.2.1 async-timeout-4.0.3 attrs-23.2.0 bcrypt-4.1.2 black-24.1a1 certifi-2023.11.17 cffi-1.16.0 click-8.1.7 cmarkgfm-0.6.0 compliance-trestle-2.4.0 cryptography-41.0.7 datamodel-code-generator-0.25.2 defusedxml-0.7.1 dnspython-2.4.2 email-validator-2.1.0.post1 et-xmlfile-1.1.0 exceptiongroup-1.2.0 frozenlist-1.4.1 furl-2.1.3 genson-1.2.2 h11-0.14.0 httpcore-1.0.2 httpx-0.26.0 idna-3.6 ilcli-0.3.2 inflect-5.6.2 isort-5.13.2 multidict-6.0.4 mypy-extensions-1.0.0 openpyxl-3.1.2 orderedmultidict-1.0.1 orjson-3.9.10 packaging-23.2 paramiko-3.4.0 pathspec-0.12.1 platformdirs-4.1.0 pycparser-2.21 pydantic-1.10.13 pynacl-1.5.0 python-dotenv-1.0.0 python-frontmatter-1.0.1 pyyaml-6.0.1 ruamel.yaml-0.18.5 ruamel.yaml.clib-0.2.8 six-1.16.0 sniffio-1.3.0 toml-0.10.2 tomli-2.0.1 typing-extensions-4.9.0 yarl-1.9.4
WARNING: You are using pip version 21.2.4; however, version 23.3.2 is available.
You should consider upgrading via the '/home/degenaro/venv.trestle.pip/bin/python -m pip install --upgrade pip' command.

**Comment by degenaro** [comment]
To get past this for the moment, I did:

(venv.trestle.pip) degenaro:~$ pip uninstall requests
WARNING: Skipping requests as it is not installed.
(venv.trestle.pip) degenaro:~$ pip install requests
Collecting requests
  Using cached requests-2.31.0-py3-none-any.whl (62 kB)
Requirement already satisfied: certifi>=2017.4.17 in ./venv.trestle.pip/lib/python3.9/site-packages (from requests) (2023.11.17)
Requirement already satisfied: idna<4,>=2.5 in ./venv.trestle.pip/lib/python3.9/site-packages (from requests) (3.6)
Collecting urllib3<3,>=1.21.1
  Downloading urllib3-2.1.0-py3-none-any.whl (104 kB)
     |████████████████████████████████| 104 kB 2.6 MB/s 
Collecting charset-normalizer<4,>=2
  Downloading charset_normalizer-3.3.2-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (142 kB)
     |████████████████████████████████| 142 kB 11.5 MB/s 
Installing collected packages: urllib3, charset-normalizer, requests
Successfully installed charset-normalizer-3.3.2 requests-2.31.0 urllib3-2.1.0
WARNING: You are using pip version 21.2.4; however, version 23.3.2 is available.
You should consider upgrading via the '/home/degenaro/venv.trestle.pip/bin/python -m pip install --upgrade pip' command.

(venv.trestle.pip) degenaro:~$ trestle version
Trestle version v2.4.0 based on OSCAL version 1.0.4

**Comment by degenaro** [comment]
Turns out that trestle has direct dependency on requests in `core/remote/cache.py`. However this dependency is not specified in setup.cfg. By luck in the past, some other dependency must have depended on requests thus satisfying trestle's direct need.'

To fix, since trestle has a direct need for requests, this must be properly specified in setup.cfg.

**Comment by degenaro** [comment]
Fixed by #1488

**Comment by deenine** [comment]
thanks @degenaro
